pyScienceMode2/rehastim_interface.py

```diff
@@ -371,3 +367,527 @@ def get_motomed_torque(self) -> float:
+
+        ml_update = sciencemode.ffi.new("Smpt_ml_update*")
+        ml_update.packet_number = self.get_next_packet_number()
+
+        #  Check if points are provided for each channel stimulated
+        for channel in upd_list_channels:
+            if not channel.is_pulse_symmetric(safety=safety):
+                raise ValueError(
+                    f"Pulse for channel {channel._no_channel} is not symmetric. "
+                    f"Please put the same positive and negative current."
+                )
+            if not channel.list_point:
+                raise ValueError(
+                    "No stimulation point provided for channel {}. "
+                    "Please either provide an amplitude and pulse width for a biphasic stimulation."
+                    "Or specify specific stimulation points.".format(channel._no_channel)
+                )
+            channel_index = channel._no_channel - 1
+            ml_update.enable_channel[channel_index] = True
+            ml_update.channel_config[channel_index].period = channel._period
+            ml_update.channel_config[channel_index].ramp = channel._ramp
+            ml_update.channel_config[channel_index].number_of_points = len(channel.list_point)
+            for j, point in enumerate(channel.list_point):
+                ml_update.channel_config[channel_index].points[j].time = point.pulse_width
+                ml_update.channel_config[channel_index].points[j].current = point.amplitude
+
+        if not sciencemode.lib.smpt_send_ml_update(self.device, ml_update):
+            raise RuntimeError("Failed to start stimulation")
+        self.log(
+            "Stimulation started",
+            "Command sent to rehastim: {}".format(self.TypeRehap24(sciencemode.lib.Smpt_Cmd_Ml_Update).name),
+        )
+        self._get_last_ack()
+
+        # This code is used to set the stimulation duration
+        total_time = 0
+        while total_time < stimulation_duration:
+            self._get_current_data()
+            self.check_stimulation_errors()
+            sleep_time = min(1, stimulation_duration - total_time)
+            time.sleep(sleep_time)
+            total_time += sleep_time
+            self._get_last_ack()
+        self.stimulation_started = True
+
+    def update_stimulation(
+        self, upd_list_channels: list, stimulation_duration: int | float = None, safety: bool = True
+    ):
+        """
+        Update the ml stimulation on the device with new channel configurations.
+
+        Parameters
+        ----------
+        upd_list_channels : list
+            Channels to stimulate.
+        stimulation_duration : int | float
+            Duration of the updated stimulation in seconds.
+        safety : bool
+            Set to True if you want to check the pulse symmetry. False otherwise.
+        """
+
+        self.start_stimulation(upd_list_channels, stimulation_duration, safety)
+
+    def stop_stimulation(self):
```

Comment:
stop_stim or end_stim ?

Reply:
is there the pause function here ?

Reply:
Renamed to end_stimulation. There is no pause but an update function (no need to initialize again to modify stimulation parameters).